components/org.wso2.carbon.identity.api.user.recovery/org.wso2.carbon.identity.rest.api.user.recovery.v2/src/main/java/org/wso2/carbon/identity/rest/api/user/recovery/v2/impl/core/utils/RecoveryUtil.java

```diff
@@ -0,0 +1,342 @@
+            throw new APIError(status, errorResponse);
+        }
+        return url;
+    }
+
+    /**
+     * Builds API error to be thrown.
+     *
+     * @param errorCode Error code.
+     * @param errorMessage Error message.
+     * @param errorDescription Error description.
+     * @param status HTTP status.
+     * @return APIError object which contains the error description.
+     */
+    public static APIError handleException(String errorCode, String errorMessage, String errorDescription,
+                                           Response.Status status) {
+
+        ErrorResponse errorResponse = buildErrorResponse(errorCode, errorMessage, errorDescription);
+        return new APIError(status, errorResponse);
+    }
+
+    /**
+     * Builds API error to be thrown for server errors.
+     *
+     * @param e Identity Recovery Exception.
+     * @param errorCode Error code.
+     * @param errorMessage Error message.
+     * @param errorDescription Error description.
+     * @param status HTTP status.
+     * @return APIError object which contains the error description.
+     */
+    public static APIError handleInternalServerError(IdentityRecoveryException e, String errorCode, String errorMessage,
+                                           String errorDescription, Response.Status status) {
+
+        ErrorResponse errorResponse = buildServerErrorResponse(e, errorCode, errorMessage, errorDescription);
+        return new APIError(status, errorResponse);
+    }
+
+    /**
+     * Returns a new PreconditionFailedException.
+     *
+     * @param tenantDomain  Tenant domain.
+     * @param description   Description of the exception.
+     * @param code          Error code.
+     * @param resetCode     Reset code given to the user by confirmation API.
+     * @param correlationId Correlation Id.
+     * @return A new PreconditionFailedException with the specified details as a response.
+     */
+    public static PreconditionFailedException buildRetryPasswordResetObject(String tenantDomain, String description,
+                                                                            String code, String resetCode,
+                                                                            String correlationId) {
+
+        // Build next API calls.
+        ArrayList<APICall> apiCallsArrayList = new ArrayList<>();
+        apiCallsArrayList.add(RecoveryUtil
+                .buildApiCall(APICalls.RESET_PASSWORD_API.getType(), Constants.RelationStates.NEXT_REL,
+                        buildURIForBody(tenantDomain, APICalls.RESET_PASSWORD_API.getApiUrl(),
+                                Constants.ACCOUNT_RECOVERY_ENDPOINT_BASEPATH), null));
+        RetryErrorResponse retryErrorResponse = buildRetryErrorResponse(
+                Constants.STATUS_PRECONDITION_FAILED_MESSAGE_DEFAULT, code, description, resetCode, correlationId,
+                apiCallsArrayList);
+        if (log.isDebugEnabled()) {
```

Comment:
`isDebugEnabled()` check is required only when building the log message is costly when it is not logged. no need to do this check if there are no string operations

Reply:
Fixed with 8a5d4f0274d4bc026253b69400b0714ad6acf7a7